ui/app/clinical/consultation/views/treatmentSections/addTreatment.html

```diff
@@ -242,12 +242,16 @@ <h3 class="orderSet-additional-info" ng-show="treatmentConfig.showAdditionalInfo
                     </div>
                     <div class="field-value">
                         <button id="SOS" type="button" class="grid-row-element fl"
-                                ng-class="{active: treatment.asNeeded}" ng-click="toggleAsNeeded(treatment)"
+                        ng-if="!treatmentConfig.isACheckBox()"        
+                        ng-class="{active: treatment.asNeeded}" ng-click="toggleAsNeeded(treatment)"
                                 accesskey="{{ ::treatmentConfig.translate('asNeededButtonAccessKey', 'MEDICATION_AS_NEEDED_ACCESS_KEY') }} ">
                             <i class="fa fa-ok"></i>
                         <span ng-bind-html="::treatmentConfig.translate('asNeededButton', 'MEDICATION_AS_NEEDED_BUTTON_PLACEHOLDER')">
                         </span>
                         </button>
+                        <input type="checkbox" id="checkBoxAsNeeded" ng-if="treatmentConfig.isACheckBox()" ng-click="toggleAsNeeded(treatment)"
+                        ng-checked="treatment.asNeeded"
+                        style="min-width: 0px;width: 20px;transform: scale(1);height: 20px;">
```

Comment:
Can we create a class for this, instead of adding inline styling?

Reply:
changes have been made